### DISCUSSION TITLE
Add Thread-Safe collection types in F#

### DISCUSSION THREAD

**Discussion by Xyncgas** [fsharp/fslang-suggestions#1239] [Library]
In C#, another product from microsoft, both being in the .NET Eco system with F#, has library types like ```System.Collections.Concurrent.ConcurrentStack<T>```

In f# we can easily make a list just by typing ```[]```, or ```let myList :list = []```, but we don't have a thread safe list, so maybe we should add one

maybe we can just do :
```
let myList =
    threadsafe []
````

**Comment by vzarytovskii** [comment]
Please, write a proper suggestion for that, or find an existing one. 
Thanks
